Feat det obj label

AbimbolaOO opened the pull request:
This commit introduces the option for the user to remove the label, confidence score, bounding box from output results.
This makes deepstack SDK suit larger audience use cases or situations where they do not want the bounding boxes, label, or confidence score shows. 
This was achieved by introducing three keywords arguments to "detectObject", "detectObjectVideo",  "drawResponse" methods and functions in the detection.py and viz.py files. Also, three positional arguments were added to "saveResponse" in the viz.py file

AbimbolaOO committed:
feat(deepstack_sdk): option to remove label and and bounding box

This commit makes use specify if they want to to remove bounding or
label, they can also remove both if they want to by specifying the
required key word arguement as seen in the demo.py file attached.

AbimbolaOO committed:
feat(deepstack_sdk): Option to remove confidence value

With this commit user now have the option to remove just the confidence
level from the label.

OlafenwaMoses commented on deepstack_sdk/viz.py:
```diff
@@ -51,7 +52,7 @@ def drawResponse(image,response,output_font=cv2.FONT_HERSHEY_SIMPLEX,output_font
                 image_arr = cv2.putText(
                     img=image_arr,
-                    text=txt + " ( " + str(100*obj.confidence)+"% )",
+                    text=(txt + (" ( " + str(100*(obj.confidence))+"% )" if show_conf else '' ))if show_label else None,
```
- Add a space between `'' ))` and `if` in this line
- I think the value `text` should be populated outside of/before the **putText** function. The current implementation conditions that label must be enabled before confidence can show. On the contrary, people should be able to (1) display only label, (2) display only confidence, (3) display none of the two.

That case, the implementation should look like this.
```suggestion
                    text = ""
                    text += txt if show_label
                    text += " ( " + str(100*obj.confidence)+"% )" if show_conf
                    
                    image_arr = cv2.putText(
                    img=image_arr,
                    text=text,
```

OlafenwaMoses commented on deepstack_sdk/viz.py:
```diff
@@ -51,7 +52,7 @@ def drawResponse(image,response,output_font=cv2.FONT_HERSHEY_SIMPLEX,output_font
                 image_arr = cv2.putText(
                     img=image_arr,
-                    text=txt + " ( " + str(100*obj.confidence)+"% )",
+                    text=(txt + (" ( " + str(100*(obj.confidence))+"% )" if show_conf else '' ))if show_label else None,
```
For displaying object confidence, I might suggest we round the value to 2 decimal places ( long decimal values doesn't look great on drawn images ). For example, we use `str(round(obj.confidence, 2))` instead of the current `obj.confidence`.

@johnolafenwa thoughts?

johnolafenwa replied:
@OlafenwaMoses , rounding the confidence to 2 dp would be great for visualization. Totally support it

OlafenwaMoses commented on demo.py:
```diff
@@ -0,0 +1,17 @@
+# from deepstack_sdk import ServerConfig, Detection
```
Please remove this file. Instead, update the existing test codes in `test/` and add the new parameters to them. You can set them all to `True` for all the samples (but the whoever view the sample will understand they can use the parameters to turn off some visualization)

OlafenwaMoses reviewed (changes requested):
Please add the following changes
- Extend this changes to **detectFace()**, **detectFaceVideo()**, **recognizeFace()** and **recognizeFaceVideo()** functions too.
- Update the sample codes in the **README.md** to include the new parameters set as `True`

johnolafenwa commented on deepstack_sdk/viz.py:
```diff
@@ -5,11 +5,11 @@
+    frame = drawResponse(image,response, show_label, draw_bounding_box, show_conf, output_font,output_font_color)
     cv2.imwrite(output,frame)
 
-def drawResponse(image,response,output_font=cv2.FONT_HERSHEY_SIMPLEX,output_font_color=(0,146,224)):
```
Good work. The ```show_label,draw_bounding_box, show_conf``` should be positioned after ```output_font_color```, this would ensure existing code that calls this function with positional args, would not break.

johnolafenwa commented on deepstack_sdk/viz.py:
```diff
@@ -5,11 +5,11 @@
     cv2.imwrite(output,frame)
 
-def drawResponse(image,response,output_font=cv2.FONT_HERSHEY_SIMPLEX,output_font_color=(0,146,224)):
+def drawResponse(image,response, show_label=True, draw_bounding_box=True, show_conf=True, output_font=cv2.FONT_HERSHEY_SIMPLEX,output_font_color=(0,146,224)):
```
The ```show_label=True, draw_bounding_box=True, show_conf=True```, should be positioned after ```output_font_color```

AbimbolaOO committed:
feat(deepstack_sdk): Options to remove label, conf scores, bounding box

This commit added options to remove label confidence score and bounding
boxes to faceRecognition and also added the ability to remove either the
label or the confidence score in the objectDetection code.

AbimbolaOO committed:
chore(README.md): update the code to include keyword args

added the following keywords args draw_bounding_box=True,
show_label=True, show_conf=True to the Object Detection code snippets
and Face Recognition  code snippet
